### PR TITLE
feat: add ws option support for "buildIdentifyThrottler"

### PR DESCRIPTION
**Pull request by Deivu**
**Please describe the changes this PR makes and why it should be merged:**
Adds support for custom identify throttlers

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

**Review by didinele — APPROVED** [review]

**Review by sdanialraza — APPROVED** [review]

**Review by Qjuh — CHANGES_REQUESTED** [review]
Can you add documentation on `ClientOptions#ws` please? In `src/util/Options.js`.

**Comment by Deivu** [comment]
> Can you add documentation on `ClientOptions#ws` please? In `src/util/Options.js`.

Resolved

File: packages/discord.js/src/util/Options.js [hunk 1]
```diff
@@ -67,6 +67,17 @@ const { version } = require('../../package.json');
  * @returns {IShardingStrategy} The strategy to use for sharding
  */
 
+/**
+ * A function to change the concurrency handling for shard identifies of this manager
+ * ```js
+ * const concurrency = 1;
+ * (manager) => new SimpleIdentifyThrottler(concurrency);
```
Comment (Qjuh) [inline packages/discord.js/src/util/Options.js:74]: Please adjust the example to reflect the default used in `@discordjs/ws` and not a hardcoded concurrency of 1.
https://github.com/discordjs/discord.js/blob/d26e022afcc84448bb91deb4dd621aea20df6617/packages/ws/src/utils/constants.ts#L32-L35

**Review by Qjuh — CHANGES_REQUESTED** [review]

**Review by 7xa5h — APPROVED** [review]

**Comment by Deivu** [comment]
Resolved

**Review by Qjuh — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**